Cookbook/Technical-Documents/XcodeTips.md

```diff
@@ -1,20 +1,70 @@
-## Tips and tricks
+# Xcode tips and tricks
 
-### Errors when pulling code/switching branches
+## Keybindings
+As we spend so much time in our ~~favourite~~ editor, it's essential to learn keybindings to decrease content switching and maintain focus.
+
+#### Navigating and finding things
+As long as the project is indexed, Xcode lets you jump to anything from anywhere.
+
+- **Cmd-Shift-0** to open the Apple documentation window.
+- **Cmd-Ctrl-[** or **]** to cycle between targets.
+
+##### Find symbols, files or text
+- **Cmd-Shift-O** and type anything to quickly jump to files or symbols. **Enter** to open, **Opt-Enter** to open in the assistant editor.
+- **Cmd-Ctrl-J** (as in “jump”) or **Cmd-Ctrl-Click** on a symbol to go to definition. Add **Opt** to open in assistant editor.
```

Comment:
You can also change Xcode settings so that option opens in a new tab instead of the assistant editor (which I personally did as I use tabs more than assistant)

Reply:
Additional tip:

Adding `Opt` + `Shift` to the shortcuts will show a floating panel allowing you to choose where you want to open the file: replace the content of an existing tab, open in a new tab, in the assistant, in a new window...

![image](https://user-images.githubusercontent.com/216089/62642510-2ae4d580-b946-11e9-9a8d-dfcf8b823831.png)

 - `Cmd-Shift-O` + type anything + `Opt-Shift-Enter` will show you the panel to choose where to open the selected file
 - `Cmd-Ctrl-Opt-Shift-J` (yeah, that's a lot of modifier keys, but once  you're used to Cmd-Ctrl-J and you know that adding Opt-Shift adds that extra behavior, you get used to it) will show the panel to choose where to open the symbol definition
 - `Cmd-Ctrl-Opt-Shift-Click` on the symbol will do the same

---

(In the same topic, Cmd-J will bring the same magic floating panel allowing you to move focus to wherever you want, though I use that one way less since it's usually faster to just click on a tab or editor to move focus than Cmd-J)

---

Also, I actually have configured my Xcode to inverse the behavior with and without Ctrl: I only need Cmd-Click instead of Cmd-Ctrl-Click to jump to definition (and Cmd-Ctrl-Click shows the dropdown menu that you'd otherwise have with Cmd-Click).

Since I use "Jump to definition" way more than the other options in the menu, I prefer that setup and avoid having to add `Ctrl` to use it :wink:

![image](https://user-images.githubusercontent.com/216089/62642729-a8a8e100-b946-11e9-8f56-db5a1e126b87.png)

Reply:
Added in https://github.com/Babylonpartners/ios-playbook/pull/205/commits/e8df3dcc2d00c1c57a604501ef5be3709c010efc